sklearn/linear_model/logistic.py

```diff
@@ -1561,64 +1556,67 @@ def fit(self, X, y, sample_weight=None):
 
         X, y = check_X_y(X, y, accept_sparse='csr', dtype=np.float64,
                          order="C")
-
+        check_classification_targets(y)
         if self.solver == 'sag':
             max_squared_sum = row_norms(X, squared=True).max()
         else:
             max_squared_sum = None
 
-        check_classification_targets(y)
+        # Encode for string labels
```

Comment:
`enc_labels` (now `encoded_labels`) refers to the encoded labels that are being considered. (by being considered I mean, the ones that are finally used to train the model, as for binary problems the higher label alone is retained hence differs from `range(len(classes))`).

Ref [this line](https://github.com/scikit-learn/scikit-learn/pull/5874/files#diff-1452d97adf1a0b794930e405aa58d64bR1598)